gr-filter/lib/ichar_decimator_impl.cc

```diff
@@ -0,0 +1,71 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2020 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "ichar_decimator_impl.h"
+#include <gnuradio/io_signature.h>
+
+namespace gr {
+namespace filter {
+
+ichar_decimator::sptr ichar_decimator::make(int decimation)
+{
+    return gnuradio::get_initial_sptr(new ichar_decimator_impl(decimation));
+}
+
+/*
+ * The private constructor
+ */
+ichar_decimator_impl::ichar_decimator_impl(int decimation)
+    : gr::sync_decimator("ichar_decimator",
+                         gr::io_signature::make(1, 1, sizeof(char)),
+                         gr::io_signature::make(1, 1, sizeof(char)),
+                         decimation)
+{
+    d_doubledecimation = 2 * decimation;
+
+    // Make sure we work with pairs of bytes (I and Q as byte)
+    gr::block::set_output_multiple(2);
```

Comment:
should this be `d_doubledecimation` instead, to make sure in `work` that there are an integer number of decimations? Otherwise I believe that there could be glitches where more or fewer than `decimation` complex<char> are dropped.